chain/network/src/peer_manager/tests/connection_pool.rs

```diff
@@ -310,3 +320,93 @@ async fn invalid_edge() {
         }
     }
 }
+
+async fn test_signed_owned_ip_address(expected_closing_reason: ClosingReason, wrong_ip_address: &Option<std::net::IpAddr>, wrong_node_key: &Option<near_crypto::SecretKey>) {
```

Comment:
Things should become simpler if you just accept the IpAddr and NodeKey to use directly instead of Options. The tests below can choose to pass either the correct one or an incorrect one.

Reply:
No it doesnt work, cause you need to set up the streams in order to set the IpAddr. so the correct IpAddr cant be passed in.